---
title: "Library preparation for 16S library"
author: "Zuzana Blahovska"
date: "5/25/2021"
output: html_document
---

## DNA extraction
1.	Take samples from -80 degree into liquid nitrogen
2.	Crush them by Precellys machine (6200 rpm/2x30s) - Cool samples between runs in liquid nitrogen. (Note from Ke: crush until the materials in tube are in powder shape, sometimes it needs 4-6 x 30s; because there is bias between tissue of different plant species and the amount of tissue in the tube is different, the crush runs can adjust according to the amount of materials inside the lysing matrix tubes. Previously I use mortar grinding by hand to make sure the material were completely crushed into powder, some people in Cologne also did this while others think it is not necessary.)
3.	At room temperature place the samples on rack and add 978 µL sodium phosphate and 122 µL MT buffer (Note from Ke: do this step fast to avoid material melt )
4.	Place the samples in Precellys machine  (6200 rpm/2x30s)
5.	Centrifuge samples 15 min -14 000x g
6.	Meanwhile pipette 250 µLPPS in 2 mL micro centrifuge tubes
7.	After centrifugation transfer 900 µL supernatant into 2 mL micro centrifuge tubes containing PPS
8.	Mix 10 by inverting
9.	Centrifuge  samples (5 min - 14 000 x g)
10.	meanwhile resuspend Binding matrix by shaking the bottle and pipette 900 µL into 2 mL micro centrifuge tubes 
11.	After centrifugation pipette 900 µL of  supernatant into 2 mL tubes with Binding matrix
12.	Place samples on a rotator for 3 min 35 rpm ( program C2) (Note from Ke: or shake by hand for 3 min)
13.	Remove samples from rotator and let the matrix settle for 10 min
14.	Remove and discard 2x 550 µL of  supernatant and with the same tip mix the remaining matrix 
15.	Pipette the resuspended matrix on the Spin filter and centrifuge for (1 min -14 000x g)
16.	Empty the catch tube and add 500 µLSEWS-M with added EtOH and resuspend the matrix
17.	Centrifuge (1 min-14000 x g) and empty the catch tube
18.	Spin tubes for (2 min -14 000 x g) to dry
19.	Place the spin filter in a new 1.5 mL tubes and let air dry for 5 min
20.	Add 50 µL of  nuclease free water
21.	mix the matrix by stirring with the tip - no pipetting
22.	Close the filter and incubate in the Thermomixer for 5 min 55 degree
23.	Centrifuge (2 min -14 000 x g)  to elute the DNA
24.	Elute DNA can be placed back on Spin filter and eluted again
25.	Store in the fridge or freezer

## Picogreen measurement
[Link to picogreen worksheet](https://labbook.au.dk/download/attachments/134359528/picogreen_worksheet.xlsx?version=1&modificationDate=1611074984526&api=v2)

Dilute the samples if possible to 3.5 ng/uL, if not use them undiluted.

## Library preparation

### PCR 1

#### Bacterial 16S

Components | 1x [µL]
------------- | -------------
buffer (Taq complete, w MgCl2) | 2.5
BSA (3%) | 2.5
dNTPs (20 mM) | 0.25
P_799F (10mM, PM_12600) | 0.75
P_1192R (10mM, PM_12599) | 0.75
VWR-Taq pol	| 0.4
H2O (use LC H2O) | 4.85
spike | 10
master mix volume | 22
DNA template | 3
Total volume | 25

Depending on the nature of the samples use spike either 10^3, 10^5 or without spike. For example, if you have samples grown on the soil, you possibly need to consider adding more spike (like 10^4 or 10^5 copies of spike) because of more complex community structure in the soil. However, if you have plants inoculated with SynCom, you may need to lower the amount of spike (10^2 or 10^3). However, in this stage, we do not have a definite answer to this. 

Now, in the common microbiome freeze, we have the spike in two different concentration with either 10^2 or 10^4 number of copies/ul. For example, if you want to have 10^3 number of copies in your individual sample, you can add 10ul spike with 10^2 number copies/ul. The more detailed information about the sequence of spike V5-V7 region and how to utilize this information after NGS sequencing, can be found in the spike document.  

#### PCR 1 program
[Link to the table](https://imgur.com/RmH3wvG)

#### PCR 1 purification

##### Purification with AM pure beads				
1. Leave AM pure beads on RT for an hour for easier handling.		
2. Re-suspend AM pure beads by mixing/shaking throughly.			
3. Into your PCR product, add 0.7 X volume of AM beads.		
4. Mix it by pipetting 10x and leave it on RT for 5 min.			
5. Place the tubes/plate on magnetic racks for 3 min to separate the beads from the solution.
6. Remove and discard the solution by pipetting.			
7. Add 150 µL (if PCR plate) or 500 µL ( tube) of 70% ethanol and incubate for 1 min. 
8. Remove and discard the solution by pipetting.			
9. Repeat the washing step.
10. Dry the beads for 5 min RT – on magnetic rack. (do not overdry since this will decrease the elution efficiency).			
11. Remove plate/tubes from magnetic rack.			
12. Add 50 µL (or less if you want to concentrate) of nuclease-free water, mix it 10 times.	13. Centrifuge (1 min-14000 g) and place the tubes in the magnetic rack.	
14. Transfer the eluent to fresh tube/plate.
15. Clean template DNA are ready for PCR2.

### PCR 2

Components | 1x [µL]
------------- | -------------
buffer (Taq complete, w MgCl2) | 2.5
BSA (3%) | 2.5
dNTPs (20 mM) | 0.25
specific F primer (10mM) | 0.75
specific R primer (2,5mM) | 3
VWR-Taq pol | 0.4
H2O (use LC H2O) | 12.6
final volume | 19
DNA template | 3

#### PCR 2 program
[Link to the table](https://imgur.com/orUmdz3)

Run 1% Agarose gel
After PCR 2 run 5 µL of sample on 1% Agarose gel for 1 hour at 120V with 1Kb marker.
[Examine your gel for correct bands and order.](https://imgur.com/a0hsfLh)

#### PCR 2 purification
Option1: Use the same procedure for AM pure beads as after PCR1. In this case, you got the mix of plant DNA and microbiome DNA for root samples.

Option2: After PCR 2, run all PCR2 product (25ul) on 1% Agarose gel for 1 hour at 120V with 1Kb marker, then use gel extractor to get the only microbiome band of each sample and transfer the gel to fresh 1.5 ml tubes. Then follow the protocol from Nucleo spin PCR and gel clean up kit to do the gel purification. From this method, the picogreen measurement below will only estimate the microbial concentration of each sample, which probably can provide a more precise guide of how to equimolarly pool all barcoded samples into one sequencing library. However, it is quite time-consuming.

#### Picogreen measurement
[Measure the concentration of samples by Picogreen.](https://labbook.au.dk/download/attachments/134359528/picogreen_worksheet.xlsx?version=1&modificationDate=1611074984526&api=v2)

## Pooling of samples: If you used AM pure beads for purification
Following steps highly depends on your gel results!

If you used AMpure beads for purification, you need to remove the plant band from root samples. 

If the plant band and microbial band are equal, you can use this set up:

For Rhizosphere sample pool them to 10 ng/µL and for root samples pool 20 ng/uL.

You can adjust the ration according to your gels pictures (it is better to use the G-box for it)

After pooling your samples run it all on run 1% Agarose gel 1 hour at 120V with 1 Kb marker and extract the microbial band. 

Clean up is done by Nucleo spin PCR and gel clean up kit.

## Pooling of samples: If you used gel purification (option2 above)
Just based on the results from picogreen to calculate the required amount of each sample to be pooled (e.g. 20ng/sample). Like the example showed below.

well | sample | PCR2 DNA conc | F_barcode | R_barcode | Pooling (20ng)
------------- | ------------- | ------------- | ------------- | ------------- | -------------
1	 | sample1 | 20.1 | B5-F-01 | B5-R-13 | 0.99
2	| sample2	| 21.7	| B5-F-01	| B5-R-14	| 0.92
3	| sample3	| 15.3	| B5-F-01	| B5-R-15	| 1.30
4	| sample4	| 13.0	| B5-F-01	| B5-R-16	| 1.53
5	| sample5	| 12.6 | B5-F-01 | B5-R-17 | 1.59

## Concentrate the library
Pool all your replicates of the library after gel clean up. The aim is to have up to 50 µL of kit purified library. If you have more, since you had to split the gel extraction into more tubes, consider purifying all the samples again, so that you end up with a low volume of all the pooled samples.

Use the same procedure for AM pure beads as after PCR1 and PCR2.

## Qubit measurement
Qubit™ dsDNA HS Assay Kit

Prepare 1 mL of working solution: 995 µL of 1xTe buffer + 5µL of  assay dye reagents

## Standards
Kit provides already prepared standards.

10µL of  Standard 1 +190 µL of working solution

10µL of  Standard 2 +190 µL of working solution

## Sample
After the library was concentrated usually is needed only 1 µL of sample for this measurement:

1 µL of sample +199 µL of working solution

If the reading cannot be perform, dilute your sample 1:10. 
